Xamarin.Forms.Platform.iOS/Renderers/ButtonRenderer.cs

```diff
@@ -170,9 +166,12 @@ protected override void SetBackground(Brush brush)
 					var backgroundImage = this.GetBackgroundImage(brush);
 					backgroundColor = backgroundImage != null ? UIColor.FromPatternImage(backgroundImage) : UIColor.Clear;
 				}
+
+				_useBackgroundBrush = true;
 			}
 
-			Control.BackgroundColor = backgroundColor;
+			if (_useBackgroundBrush)
```

Comment:
Should this be using a locally stored _defaultBackgroundColor instead of a bool check?

Basically how SetBackgroundColor is setup?

```C#
		protected override void SetBackgroundColor(Color color)
		{
			if (IsElementOrControlEmpty)
				return;
#if __MOBILE__
			if (color == Color.Default)
				Control.BackgroundColor = _defaultColor;
			else
				Control.BackgroundColor = color.ToUIColor();
#else
			Control.Layer.BackgroundColor = color == Color.Default ? _defaultColor : color.ToCGColor();
#endif
		}
```

Reply:
Yes, we already have a variable with the default color, the new bool variable only indicates if a brush is being used or not. We have two properties `Background` and `BackgroundColor` and this variable just indicate us when we can update the background using a color.

Reply:
This still feels a little bit off to me. `_useBackgroundBrush` can only ever be toggled one direction so once it's true it's always true

So, after you set the Background to something the `BackgroundColor` effectively becomes useless

```C#
        private void Button_Clicked(object sender, EventArgs e)
        {
            btnMe.Background = SolidColorBrush.Default;
            btnMe.BackgroundColor = Color.Green;
        }

        private void Button_Clicked2(object sender, EventArgs e)
        {
           // this code stops working after you've set the Background to a brush. 
            btnMe.Background = SolidColorBrush.Purple;
        }
```

Is this expected behavior?

Reply:
Shane, I totally see your feedback and I understand it. I have applied changes with a different approach with more sense.
Now, the custom renderer works and also your case works without using any internal variable etc.